processor/src/operations/crypto_ops.rs

```diff
@@ -102,23 +102,21 @@ where
     /// - new value of the node, 4 elements.
     ///
     /// To perform the operation we do the following:
-    /// 1. Update the leaf node at the specified index in the advice provider with the specified
-    ///    root, and get the Merkle path to this leaf. If `copy` is set to true, we make a copy
-    ///    of the merkle set before updating it.
+    /// 1. Update the node at the specified index in the advice provider with the specified root,
+    ///    and get the Merkle path to it. It will make a copy of the Merkle tree before updating
+    ///    it.
     /// 2. Use the hasher to update the root of the Merkle path for the specified node. For this
     ///    we need to provide the old and the new node value.
     /// 3. Verify that the computed old root is equal to the input root provided via the stack.
     /// 4. Replace the old node value with the computed new root.
     ///
     /// The Merkle path for the node is expected to be provided by the prover non-deterministically
-    /// (via merkle sets). At the end of the operation, the old node value is replaced with the
-    /// new root value computed based on the provided path. Everything else on the stack remains the
-    /// same.
-    ///
-    /// If `copy` is set to true, at the end of the operation the advice provide will keep both,
-    /// the old and the new merkle sets. Otherwise, the old merkle set is removed from the
-    /// provider.
+    /// (via the advice provider). At the end of the operation, the old node value is replaced with
+    /// the new roots value computed based on the provided path. Everything else on the stack
+    /// remains the same.
     ///
+    /// At the end of the operation, the advice provider will keep both the old and the new Merkle
+    /// trees.
```

Comment:
Let's change this to: "The original Merkle tree is cloned before the update is preformed, and thus, after the operation, the advice provider will keep track of both the old and the new trees."

Reply:
Fixed